src/fileimpl.cpp

```diff
@@ -526,7 +524,7 @@ namespace zim
     {
       std::ostringstream msg;
       msg << "unknown mime type code " << idx;
-      throw std::runtime_error(msg.str());
+      throw ZimFileFormatError(msg.str());
```

Comment:
`Item::getMimetype()` has no access to the number of mimetype in the file, so it cannot do the check.
`FileImpl::getMimeType` can only be called by internal libzim method (it is not part of the public api). Actually, it is called only by `Item::getMimetype`, if the index is invalid it is because the zimfile is incorrect.